src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java

```diff
@@ -48,125 +39,163 @@
+  private final By pauseOverlaySelector;
+  private final By closeButtonSelector;
+  private final By progressBarSelector;
+  private final By replayOverlaySelector;
 
-  private boolean playing;
+  private boolean mobile;
 
-  private boolean muted;
+  public AutoplayVuap(WikiaWebDriver driver, String slot, String adIframeId) {
+    this(driver, slot, By.id(adIframeId), false);
+  }
 
-  public AutoplayVuap(WikiaWebDriver driver, String slot, String videoIframeSelector) {
+  public AutoplayVuap(WikiaWebDriver driver, String slot, By adIframeSelector, Boolean mobile) {
     this.driver = driver;
     this.wait = new Wait(driver);
 
     this.slot = slot;
-    this.videoIframeSelector = videoIframeSelector;
-    this.playing = true;
-    this.muted = true;
+    this.imaBridgeSelector = By.cssSelector("#" + slot + " .video-player iframe[src*='imasdk']");
+    this.adIframeSelector = adIframeSelector;
+    this.mobile = mobile;
+
+    speakerSelector = By.cssSelector(String.format(SPEAKER_SELECTOR_FORMAT, slot));
+    pauseOverlaySelector = By.cssSelector(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot));
+    closeButtonSelector = By.cssSelector(String.format(CLOSE_BUTTON_SELECTOR_FORMAT, slot));
+    progressBarSelector = By.cssSelector(String.format(CURRENT_TIME_SELECTOR_FORMAT, slot));
+    replayOverlaySelector = By.cssSelector(String.format(REPLAY_BUTTON_SELECTOR_FORMAT, slot));
   }
 
   public void mute() {
-    if (!muted) {
-      clickElement(String.format(SPEAKER_SELECTOR_FORMAT, slot));
-      muted = true;
+    if (!isMuted()) {
+      wait.forElementClickable(speakerSelector).click();
     }
   }
 
   public void unmute() {
-    if (muted) {
-      clickElement(String.format(SPEAKER_SELECTOR_FORMAT, slot));
-      muted = false;
+    if (isMuted()) {
+      wait.forElementClickable(speakerSelector).click();
     }
   }
 
-  public void pause() {
-    if (playing) {
-      clickElement(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot));
-      playing = false;
-    }
+  public void togglePause() {
+    wait.forElementClickable(pauseOverlaySelector).click();
   }
 
   public void play() {
-    if (!playing) {
-      clickElement(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot));
-      playing = true;
+    if (isPaused()) {
+      togglePause();
+    } else if( !this.mobile ) {
+      clickOnArea(2);
+    } else {
+      clickOnArea(4);
     }
   }
 
-  public void playVuapVideo() {
-    if (playing) {
-      clickOnClickArea2();
-      playing = true;
+  public void playVideoForFewSeconds() {
+    playVideoFor(Duration.ofSeconds(2));
+  }
+
+  public void playVideoFor(Duration duration) {
+    play();
+
+    try {
+      Thread.sleep(duration.toMillis());
+    } catch (InterruptedException x) {
+      // ignore this exception
     }
+
+    togglePause();
+  }
+
+  private boolean isPaused() {
+    return isVideoPaused() && driver.findElement(pauseOverlaySelector).isDisplayed();
+  }
+
+  private Boolean isVideoPaused() {
```

Comment:
There is a difference (`isVideoPaused` is true when it's paused or when it isn't even started), but you are right, names are not precise enough